File: web/frontend/src/api/status.ts
// API client for the picoclaw web backend.

const BASE_URL = ""

interface StatusResponse {
  status: string
  version: string
  uptime: string
}

async function request<T>(path: string, options?: RequestInit): Promise<T> {
  const res = await fetch(`${BASE_URL}${path}`, options)
  if (!res.ok) {
    throw new Error(`API error: ${res.status} ${res.statusText}`)
  }
  return res.json() as Promise<T>
}

export async function getStatus(): Promise<StatusResponse> {
  return request<StatusResponse>("/api/status")
}

File: web/backend/api/status.go
package api

import (
	"encoding/json"
	"net/http"
	"time"

	"github.com/sipeed/picoclaw/web/backend/model"
)

// startTime records when the server was started, used to calculate uptime.
var startTime = time.Now()

// Version is set at build time via -ldflags.
var Version = "dev"

// handleStatus returns the current server status, version, and uptime.
//
//	GET /api/status
//	Response: 200 OK
//	{
//	  "status": "online",
//	  "version": "dev",
//	  "uptime": "2h30m15s"
//	}
func (h *Handler) handleStatus(w http.ResponseWriter, r *http.Request) {
	resp := model.StatusResponse{
		Status:  "online",
		Version: Version,
		Uptime:  time.Since(startTime).Round(time.Second).String(),
	}
	json.NewEncoder(w).Encode(resp)
}
